src/main/java/tuteez/model/person/lesson/Lesson.java

```diff
@@ -164,14 +164,14 @@ public static boolean isClashingWithOtherLesson(Lesson firstLs, Lesson secondLs)
      * @return {@code true} if any two lessons in the set have a timing conflict,
      *         {@code false} otherwise.
      */
-    public static boolean containsClashes(List<Lesson> lessons) {
-        List<Lesson> lessonList = new ArrayList<>(lessons); // Convert to list for easy indexing
+    public static boolean hasClashingLessonWithinList(List<Lesson> lessons) {
```

Comment:
Perhaps this method can be further abstracted as it contains 3 levels of nesting